### DISCUSSION TITLE
S3Error: Resource requested is unreadable, please reduce your request rate

### DISCUSSION THREAD

**Discussion by docwarems** [minio/minio#16341] [Q&A] [answered]
I'm using MinIO with nodejs. During a putObject() operation I once got this error :
"S3Error: Resource requested is unreadable, please reduce your request rate"

I've recently started learning nodejs stream processing from scratch. I learned about stream.Writable where the write() method returns a boolean. And if it returns false, you have to pause the processing and resume it on the "drain" event. By this you can avoid overloading the outputstream and avoid backpressure issues.
But I cannot see how the minIO putObject() method supports a functionality like this. It returns void. The callback returns the error above. Is a retry on the putObject() the only thing I can do now?

"reduce your request rate" ist a little vague. I don't want to include unnecessary waits into my stream processing. In a multitenant environment it's also of course not only up to me what pressure is currently done on the MinIO server, 
How can I cope with that problem?

**Comment by klauspost** [comment, marked as answer]
Check your server logs. The error should only occur if there is a problem with your setup or it is heavily loaded.

**Reply by docwarems** [reply]
Thank you I will forward the issue to our server admin to let him check the logs. I'll come back with a reaction.

**Reply by docwarems** [reply]
Yes, the MinIO server log complains about a disk unavailable. Obviously some kind of cluster problem.
Thanks for your help.
